packages/connector-tencent-sms/src/index.ts

```diff
@@ -0,0 +1,120 @@
+import type {
+  CreateConnector,
+  GetConnectorConfig,
+  SendMessageFunction,
+  SmsConnector,
+} from '@logto/connector-kit';
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorType,
+  validateConfig,
+} from '@logto/connector-kit';
+import { assert } from '@silverhand/essentials';
+import { HTTPError } from 'got';
+
+import { sendSMSRequest } from '@/http';
+import type { TencentSmsConfig } from '@/schema';
+import { SmsConfigGuard, TencentErrorResponse } from '@/schema';
+
+import { defaultMetadata } from './constant';
+
+function sendMessage(getConfig: GetConnectorConfig): SendMessageFunction {
+  // eslint-disable-next-line complexity
+  return async (data, inputConfig) => {
+    const { to, type, payload } = data;
+    const config = inputConfig ?? (await getConfig(defaultMetadata.id));
+    validateConfig<TencentSmsConfig>(config, SmsConfigGuard);
+    const { accessKeyId, accessKeySecret, signName, templates, sdkAppId, region } = config;
+    const template = templates.find(({ usageType }) => usageType === type);
+
+    assert(
+      template,
+      new ConnectorError(
+        ConnectorErrorCodes.TemplateNotFound,
+        `Cannot find template for type: ${type}`
+      )
+    );
+
+    try {
+      const httpResponse = await sendSMSRequest(template.templateCode, [payload.code], to, {
+        secretId: accessKeyId,
+        secretKey: accessKeySecret,
+        sdkAppId,
+        region,
+        signName,
+      });
+
+      const { body: responseData } = httpResponse;
+
+      try {
+        TencentErrorResponse.parse(responseData);
+        throw new HTTPError(httpResponse);
+      } catch {
+        // Not http error
+      }
```

Comment:
nit: `XXX.safeParse` instead of try-catch? Avoid deeply nested code blocks. 

Also Just curious, what would be the use case for this, the original response is 2XX but with an invalid response body and we throw an HTTPError instead?   Should we throw a ConnectorError type here ? cc: @wangsijie

Reply:
The `HTTPError` thrown will be caught locally and will not be thrown externally

Reply:
Trying to list all the exception cases:
1. HTTP error response non 2XX status
2. HTTP response 2XX but with a `TencentErrorResponse` response body
3. Missing SendStatusSet[0].Code === 'OK' in response?